### PR TITLE
created a user booting function; added message passing to the login

### PR DESCRIPTION
**Pull request by charlesstrange2**
https://trello.com/c/SBHNJW0M/1690-market-is-not-logging-people-out-when-logged-in-at-time-in-trader-disabling

NetMgr creates an event when it gets a 403, from someone misbehaving when connecting to an API endpoint they shouldn't
Main picks that up and logs the user out before sending them on to the Login page, with a message code that renders an appropriate  message.

### CODE REVIEW

File: src/components/Profile.vue [hunk 1]
```diff
@@ -28,7 +28,8 @@ export default {
     methods: {
         onLogout: function () {
             Store.unAuthenticate();
-            this.$router.push("/login");
+            const routeObj = { name: "login", params: { logoutReason: 200 } };
```
Comment (katjam) [inline src/components/Profile.vue:32]: what does logoutReason 200 mean? Could be made more clear here that it means "normal logout"

File: src/constants.js [hunk 2]
```diff
@@ -17,5 +17,12 @@ export default {
             " has been successfully removed from your list.",
         DELETE_VOUCHER_FAIL:
             " cannot be removed at this time, please try later.",
+        FORCED_LOGOUT:
+            "The system has logged you out, please login again in a little while.",
+        USER_LOGOUT: "You have logged out.",
+        TIMEOUT_LOGOUT: "Your session has timed out, please login again.",
+        UNKNOWN_EVENT: "Something unusual has happened.",
```
Comment (katjam) [inline src/constants.js:24]: :)

File: src/main.js [hunk 3]
```diff
@@ -37,23 +35,29 @@ const routes = [
...
-        }
+        },
     },
     { path: "/account", component: Account, meta: { auth: true } },
     { path: "/scan", component: Scan, meta: { auth: true } },
     { path: "/payment", component: Payment, meta: { auth: true } },
-    { path: "/login", component: Login, meta: { auth: false } },
+    {
+        path: "/login",
+        name: "login",
```
Comment (katjam) [inline src/main.js:56]: The others don't have names. Does it matter?

File: src/pages/Login.vue [hunk 4]
```diff
@@ -80,6 +80,32 @@ export default {
             env: Config.env,
         };
     },
+    mounted: function () {
```
Comment (katjam) [inline src/pages/Login.vue:83]: Nice!

File: src/services/netMgr.js [hunk 5]
```diff
@@ -252,6 +252,12 @@ NetMgr.axiosInstance.interceptors.response.use(
             NetMgr.setOnlineStatus(false);
         }
 
+        // is it a 403? USer tried something bad, broadcast a Logout event, someone will deal.
```
Comment (katjam) [inline src/services/netMgr.js]: Typo 
```suggestion
        // is it a 403? User tried something bad, broadcast a Logout event, someone will deal.
```

**Review by katjam — APPROVED** [review]
A tiny typo and maybe some extra doc needed on standard logout to ref the logout reason Consts but otherwise ace!

File: src/components/Profile.vue (hunk 1 above)
Comment (charlesstrange2) [reply src/components/Profile.vue:32]: I'll add a comment

File: src/main.js (hunk 3 above)
Comment (charlesstrange2) [reply src/main.js:56]: We call them by path elsewhere; oddly we need a name to do this, apparently 🤷

**Pull request merged**